multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -687,15 +687,6 @@ def fit(
             fit_called=fit_called,
         )
 
-        train_data, tuning_data = split_train_tuning_data(
```

Comment:
Moving `split_train_tuning_data ` after `infer_column_types ` would cause issues since `infer_column_types ` requires to use `tuning_data` in training. Otherwise, columns with one unique value can't be ignored. See https://github.com/autogluon/autogluon/blob/master/multimodal/src/autogluon/multimodal/data/infer_types.py#L504-L528

Reply:
Updated to resolve this. Note that it is a bit awkward to resolve this given how logic is being coupled together in the current functions, so I've added TODOs and FIXMEs where I think it can be streamlined.